android/src/main/java/io/radar/react/RNRadarModule.java

```diff
@@ -129,14 +177,19 @@ public boolean onRequestPermissionsResult(int requestCode, String[] permissions,
 
     @ReactMethod
     public void requestPermissions(boolean background, final Promise promise) {
-        PermissionAwareActivity activity = (PermissionAwareActivity)getCurrentActivity();
+        PermissionAwareActivity activity = (PermissionAwareActivity) getCurrentActivity();
         mPermissionsRequestPromise = promise;
         if (activity != null) {
-            if (Build.VERSION.SDK_INT >= 23) {
-                if (background && Build.VERSION.SDK_INT >= 29) {
-                    activity.requestPermissions(new String[] { Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_BACKGROUND_LOCATION }, PERMISSIONS_REQUEST_CODE, this);
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                if (background && Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+                    activity.requestPermissions(new String[]{
+                            Manifest.permission.ACCESS_FINE_LOCATION,
```

Comment:
I wonder if we need to start passing in `COARSE` here as well. 🤔   I bet the OS will just show the standard prompt with both `FINE` and `COARSE` even if you don't ask for `COARSE`.  I bet we're fine with the code as it is.

Reply:
Good catch. I will add it. This code is all based off the sdk implementation, which does not request coarse permissions at the moment. There is a "permission-groups" feature that we should look into for all Android-based SDKs.